website/docs/deployment.mdx

```diff
@@ -140,14 +140,182 @@ cmd /C 'set "GIT_USER=<GITHUB_USERNAME>" && yarn deploy'
 
 [GitHub Actions](https://help.github.com/en/actions) allow you to automate, customize, and execute your software development workflows right in your repository.
 
-This workflow assumes your documentation resided in `documentation` branch of your repository and your [publishing source](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) is configured for `gh-pages` branch.
+This workflow assumes your documentation resides in the `documentation` branch of your repository and your [publishing source](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) is configured for the `gh-pages` branch.
+
+Below are two approaches to deploying your docs with GitHub Actions. Follow the example based on if your deployment branch (eg `gh-pages`) is in:
+
+- The **`same`** repository as your documentation source.
+- A **`remote`** repository different from your documentation source.
+
+````mdx-code-block
+<Tabs
+  defaultValue="same"
+  values={[
+    { label: 'Same', value: 'same' },
+    { label: 'Remote', value: 'remote' }
+]}>
+<TabItem value="same">
+
+Add these two workflow files for Github Actions to your project:
+
+```yaml title=".github/workflows/documentation-push.yml"
+# Appears in various parts of the Github UI usually with the job name appended.
+# (eg Project 'Actions' tab, PR check suite status)
+name: 'Docs (Github Pages)'
+
+on:
+  # Allows you to manually trigger the workflow from the "Actions" tab.
+  # You can run the workflow against the latest commit of any branch on the project:
+  # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+  # (This is not required to use "re-run" functionality on prior workflow runs)
+  workflow_dispatch:
+  # Build docs pushed to the `documentation` branch, but only *if* the triggering
+  # commit(s) affect files within the `website` directory (docusaurus project):
+  push:
+    branches:
+      - documentation
+    paths:
+      - 'website/**'
+    # OPTIONAL: Build docs when you push a tag with `v<Major>.<Minor>` substring in the tag name.
+    # NOTE: The branch and path conditions above have no effect on the tags trigger condition:
+    tags:
+      - 'v[0-9]+.[0-9]+*'
+
+# Jobs will run shell commands from this subdirectory by default:
+defaults:
+  run:
+    working-directory: website
+
+# As this workflow example only uses these credentials once, you could instead inline them where they're used.
+env:
+  # Assign commit authorship to official Github Actions bot when pushing to the `gh-pages` branch:
+  # https://github.com/actions/checkout/issues/13#issuecomment-724415212
+  # https://api.github.com/users/github-actions%5Bbot%5D
+  GIT_USER: 'github-actions[bot]'
```

Comment:
I'm not 100% familiar with the users in github actions but isn't that done automatically?

Reply:
Isn't what done automatically? The user implied? Who would be attributed by default?

General examples I see around (including current Docusaurus gh-pages example) is to explicitly add this git user and e-mail data. As I'm using a third-party GH Action for deployment, it may have different default behaviour.

If you meant the env var names I defined, this was just me matching the property names IIRC, not intentionally overriding any existing one if there are any.

The two URLs in the comments provide information on why the more verbose values are provided. If you don't have the e-mail as I've provided, then the commit will say "github-actions" but have a light gray profile icon that links to no profile/user. When you provide the proper e-mail account for the official Github Actions user account, it will correctly link to it and display the dark profile icon associated to that account.